### PR TITLE
Improve DLL hijacking rule coverage

### PR DESCRIPTION
**Pull request by ikretz**
This PR improves the coverage of the DLL hijacking rules by adding patterns to match strings resembling various Windows commands for executing DLLs.

### CODE REVIEW

File: guarddog/analyzer/sourcecode/dll-hijacking.yml [hunk 1]
```diff
@@ -34,6 +34,24 @@ rules:
                 # a string with .dll or .so in it
                 - pattern: "..."
                 - pattern-regex: (?i).*?\.(dll|so)
+        - patterns:
+          - pattern: "..."
+          - pattern-either:
+            # See MITRE ATT&CK "System Binary Proxy Execution" techniques
+            # https://attack.mitre.org/techniques/T1218/
```
Comment (sobregosodd) [inline guarddog/analyzer/sourcecode/dll-hijacking.yml:41]: I would use (?i) in front of all regex to make the case insensitve

File: guarddog/analyzer/sourcecode/dll-hijacking.yml [hunk 2]
```diff
@@ -34,6 +34,24 @@ rules:
...
+            - pattern-regex: control(.exe)?\s*\S*.cpl
```
Comment (sobregosodd) [inline guarddog/analyzer/sourcecode/dll-hijacking.yml:42]: \s+\S+ meaning that we need at least one space and one nonspace

File: guarddog/analyzer/sourcecode/dll-hijacking.yml (hunk 1 above)
Comment (sobregosodd) [reply guarddog/analyzer/sourcecode/dll-hijacking.yml:41]: we could move this entire pattern-either block below line 22

**Review by sobregosodd — CHANGES_REQUESTED** [review]

File: guarddog/analyzer/sourcecode/dll-hijacking.yml (hunk 1 above)
Comment (ikretz) [reply guarddog/analyzer/sourcecode/dll-hijacking.yml]: Both changes made

File: guarddog/analyzer/sourcecode/dll-hijacking.yml (hunk 2 above)
Comment (ikretz) [reply guarddog/analyzer/sourcecode/dll-hijacking.yml]: Changed

**Pull request merged**